src/components/Button/Button.tsx

```diff
@@ -0,0 +1,50 @@
+import React from "react";
+import { useKnockFeed } from "../KnockFeedProvider";
+import { Spinner } from "../Spinner";
+import { ButtonSpinner } from "./ButtonSpinner";
+
+import "./styles.css";
+
+export type ButtonProps = {
+  variant: "primary" | "secondary";
+  loadingText?: string;
+  isLoading?: boolean;
+  isDisabled?: boolean;
+  isFullWidth?: boolean;
+  onClick: (e: React.MouseEvent) => void;
+};
+
+export const Button: React.FC<ButtonProps> = ({
+  variant = "primary",
+  loadingText,
+  isLoading = false,
+  isDisabled = false,
+  isFullWidth = false,
+  onClick,
+  children,
+}) => {
+  const { colorMode } = useKnockFeed();
+
+  const classNames = [
+    "rnf-button",
+    `rnf-button--${variant}`,
+    isFullWidth ? "rnf-button--full-width" : "",
+    isLoading ? "rnf-button--is-loading" : "",
+    `rnf-button--${colorMode}`,
+  ].join(" ");
+
+  const showWhileLoading = loadingText || (
+    <span className="rnf-button__button-text-hidden">{children}</span>
```

Comment:
Just curious, what's the reason we bother falling back to this `children` but with a span/css to hide it? i.e. could we just do `{isLoading ? loadingTest : children}` below?

Reply:
Let me add a comment here, but the reason we do this is so that the button width does not shift when the spinner is displayed when there's no loading text, and instead we show the spinner in the center of the button. That's why we use the `rnf-button__button-text-hidden` class to do that (which sets opacity to 0).

Basically took this from Chakras button implementation (as it's very good!)

Reply:
Aha, makes sense!